src/Indice.Features.Risk.Core/Configuration/RiskEngineBuilder.cs

```diff
@@ -24,46 +24,56 @@ public class RiskEngineBuilder
     /// <summary>Adds a new rule to the risk engine by providing an implementation of <see cref="RiskRule"/>.</summary>
     /// <typeparam name="TRule">The implementation type.</typeparam>
     /// <typeparam name="TOptions">The options associated with the rule.</typeparam>
+    /// <typeparam name="TValidator">The validator associated with the rule options.</typeparam>
     /// <param name="name">The name of the rule.</param>
     /// <returns>The instance of <see cref="RiskEngineBuilder"/>.</returns>
-    public RiskEngineBuilder AddRule<TRule, TOptions>(string name)
+    public RiskEngineBuilder AddRule<TRule, TOptions, TValidator>(string name)
         where TRule : RiskRule
-        where TOptions : RuleOptionsBase, new() {
+        where TOptions : RuleOptionsBase, new()
+        where TValidator : RuleOptionsBaseValidator<TOptions>, new() {
         CheckAndAddRuleName(name);
-        _services.AddOptions<TOptions>().BindConfiguration($"{RULE_OPTIONS_SECTION}:{name}");
+        _services.AddTransient<IValidator<TOptions>, TValidator>();
+        _services.AddOptions<TOptions>().BindConfiguration($"{Constants.RuleOptionsSectionName}:{name}");
         _services.AddTransient<RiskRule, TRule>();
         return this;
     }
 
     /// <summary>Adds a new rule to the risk engine by providing a lambda expression.</summary>
     /// <typeparam name="TOptions">The options associated with the rule.</typeparam>
+    /// <typeparam name="TValidator">The validator associated with the rule options.</typeparam>
     /// <param name="name">The name of the rule.</param>
     /// <param name="ruleDelegate">The delegate method to when a new event occurs.</param>
     /// <returns>The instance of <see cref="RiskEngineBuilder"/>.</returns>
-    public RiskEngineBuilder AddRule<TOptions>(
+    public RiskEngineBuilder AddRule<TOptions, TValidator>(
         string name,
         Func<IServiceProvider, RiskEvent, ValueTask<RuleExecutionResult>> ruleDelegate
```

Comment:
να σας πω την αλήθεια μου τη σπάει να βάζεις κανονα ξερά με ενα delegate, ίσως θα πρέπει να το φάμε και να πάμε με κανονικές κλάσεις. @ar-is . ας το αφήσουμε για την ώρα αλλά γίνεται μη συντηρήσιμος ο κώδικας του πελάτη

Reply:
πρακτικα αυτο το overload δεν το χρησιμοποιουμε καν, μπορουμε να το σβησουμε και τωρα.